contracts/utils/Create2.sol

```diff
@@ -59,8 +57,16 @@ library Create2 {
         bytes32 salt,
         bytes32 bytecodeHash,
         address deployer
-    ) internal pure returns (address) {
-        bytes32 _data = keccak256(abi.encodePacked(bytes1(0xff), deployer, salt, bytecodeHash));
-        return address(uint160(uint256(_data)));
+    ) internal pure returns (address addr) {
+        /// @solidity memory-safe-assembly
+        assembly {
+            mstore(0x00, deployer)
+            mstore8(0x0b, 0xff)
+            mstore(0x20, salt)
+            let ptr := mload(0x40)
+            mstore(0x40, bytecodeHash)
+            addr := keccak256(0x0b, 85)
+            mstore(0x40, ptr)
```

Comment:
I think this would be cleaner to read. Would it affect the stack in a way that lead to more gas usage ?
```suggestion
            let ptr := mload(0x40)
            mstore(0x00, deployer)
            mstore8(0x0b, 0xff)
            mstore(0x20, salt)
            mstore(0x40, bytecodeHash)
            addr := keccak256(0x0b, 85)
            mstore(0x40, ptr)
```